src/helics/apps/Connector.cpp

```diff
@@ -0,0 +1,1202 @@
+    if (json.isMember("aliases")) {
+        for (auto& alias : json["aliases"]) {
+            const std::string_view alias1 =
+                connections.interfaces.emplace_back(alias[0].asString());
+            const std::string_view alias2 =
+                connections.interfaces.emplace_back(alias[1].asString());
+            connections.aliases.emplace(alias1, alias2);
+        }
+    }
+    if (json.isMember("tags")) {
+        loadTags(connections, json["tags"]);
+    }
+    if (json.isMember("unknown_inputs")) {
+        for (const auto& input : json["unknown_inputs"]) {
+            connections.unknownInputs.push_back(input.asString());
+        }
+    }
+    if (json.isMember("unknown_publications")) {
+        for (const auto& pub : json["unknown_publications"]) {
+            connections.unknownPubs.push_back(pub.asString());
+        }
+    }
+    if (json.isMember("unknown_endpoints")) {
+        for (const auto& ept : json["unknown_endpoints"]) {
+            connections.unknownEndpoints.push_back(ept.asString());
+        }
+    }
+    for (auto& broker : json["brokers"]) {
+        brokerConnectionList(connections, broker);
+    }
+    for (auto& core : json["cores"]) {
+        coreConnectionList(connections, core);
+    }
+    connections.tagStrings.push_back("default");
+    connections.tagCodes.reserve(connections.tagStrings.size());
+    std::transform(connections.tagStrings.begin(),
+                   connections.tagStrings.end(),
+                   std::back_inserter(connections.tagCodes),
+                   std::hash<std::string>());
+    return connections;
+}
+
+Connector::Connector(std::vector<std::string> args):
+    App("connector", std::move(args)), core((fed) ? fed->getCorePointer() : nullptr)
+{
+    processArgs();
+}
+
+Connector::Connector(int argc, char* argv[]):
+    App("connector", argc, argv), core((fed) ? fed->getCorePointer() : nullptr)
+{
+    processArgs();
+}
+
+void Connector::processArgs()
+{
+    auto app = generateParser();
+
+    if (!deactivated) {
+        fed->setFlagOption(HELICS_FLAG_SOURCE_ONLY);
+        app->helics_parse(remArgs);
+        if (!masterFileName.empty()) {
+            loadFile(masterFileName);
+        }
+    } else if (helpMode) {
+        app->remove_helics_specifics();
+        std::cout << app->help();
+    }
+}
+
+const std::unordered_map<std::string_view, InterfaceDirection> directionNames{
+    {"from_to", InterfaceDirection::FROM_TO},
+    {"1", InterfaceDirection::FROM_TO},
+    {"FROM_TO", InterfaceDirection::FROM_TO},
+    {"to_from", InterfaceDirection::TO_FROM},
+    {"TO_FROM", InterfaceDirection::TO_FROM},
+    {"-1", InterfaceDirection::TO_FROM},
+    {"bidirectional", InterfaceDirection::BIDIRECTIONAL},
+    {"BIDIRECTIONAL", InterfaceDirection::BIDIRECTIONAL},
+    {"0", InterfaceDirection::BIDIRECTIONAL},
+    {"bi", InterfaceDirection::BIDIRECTIONAL},
+    {"BI", InterfaceDirection::BIDIRECTIONAL}};
```

Comment:
The numeric names seem like they will be a common source of mistakes. It's not really clear that they add much beyond saving a few keystrokes, and are arbitrarily assigning a numeric value to an otherwise numberless entity?

Reply:
they are the values that would get printed if the configuration were generated using enumeration values

Reply:
Is there a way those strings could get generated at compile time from the enum? One less place that could get out of sync if the enum values ever change.